Lib/test/test_httplib.py

```diff
@@ -1155,6 +1155,63 @@ def run_server():
         thread.join()
         self.assertEqual(result, b"proxied data\n")
 
+    def test_putrequest_override_validation(self):
+        """
+        It should be possible to override the default validation
+        behavior in putrequest (bpo-38216).
+        """
+        class UnsafeHTTPConnection(client.HTTPConnection):
+            def _prepare_path(self, url):
+                return url.encode('ascii')
+
+        conn = UnsafeHTTPConnection('example.com')
+        conn.sock = FakeSocket('')
+        conn.putrequest('GET', '/\x00')
+
+    def test_putrequest_override_encoding(self):
+        """
+        It should be possible to override the default encoding
+        to transmit bytes in another encoding even if invalid
+        (bpo-36274).
+        """
+        class UnsafeHTTPConnection(client.HTTPConnection):
+            def _prepare_path(self, url):
+                # Prevent CVE-2019-9740.
+                if match := client._contains_disallowed_url_pchar_re.search(
+                        url):
```

Comment:
This little hanging thing looks weird. Maybe do smth like
```python
pattern = client._contains_disallowed_url_pchar_re
if match := pattern.search(url):
```

Reply:
Or just ignore the line length _guideline_ for this one case.

Reply:
it's fine as is.  using `m` as the name instead of `match` would've been my suggestion. ;)